storage/storage_test.go

```diff
@@ -264,20 +264,24 @@ func makeLayer(t *testing.T, compression archive.Compression) (ddigest.Digest, i
 			Typeflag:   tar.TypeReg,
 		})
 		if err != nil {
-			t.Fatalf("Error writing tar header: %v", err)
+			errs <- fmt.Errorf("Error writing tar header: %v", err)
 		}
 		n, err := twriter.Write(buf)
 		if err != nil {
-			t.Fatalf("Error writing tar header: %v", err)
+			errs <- fmt.Errorf("Error writing tar header: %v", err)
 		}
 		if n != len(buf) {
-			t.Fatalf("Short write writing tar header: %d < %d", n, len(buf))
+			errs <- fmt.Errorf("Short write writing tar header: %d < %d", n, len(buf))
 		}
 		err = twriter.Flush()
 		if err != nil {
 			t.Fatalf("Error flushing output to tar archive: %v", err)
 		}
 	}()
+	err = <-errs
```

Comment:
I can’t see how this doesn’t block forever if there is no error; `errs` is never closed, and nothing is sent into it.

(See above, a good reason to dedicate a separate function to dealing with the channel properly.)